src/main/UseCases/UseCaseUpdate.java

```diff
@@ -0,0 +1,72 @@
+import java.util.*;
+
+public class UseCaseUpdate {
+
+    public void updateQuantity(String serial_num, Employee e, int modified_quantity) throws Exception {
+        // modified_quantity can be both positive and negative!
+        Item item = TempDataStorage.getItem(serial_num);
+        boolean check_authority = e.getUpdateItem();
+        if (check_authority){
+            int quantity = item.getQuantity() + modified_quantity;
+            item.setQuantity(quantity);
+        }else{
+            throw new Exception("You don't have rights to update the quantity!");
+        }
+    }
+
+    public boolean checkQuantity(String serial_num, int decreased_quantity) throws Exception{
+        // You should call this method only if the modified_quantity is negative, and you
+        // are calling this method to check if the quantity goes to negative or not.
+        Item item = TempDataStorage.getItem(serial_num);
+        int quantity = item.getQuantity();
+        if (quantity - decreased_quantity >=0){
+            return true;
+        }else{
+            throw new Exception("Quantity will be going to negative!");
+        }
+    }
+
+    public void updatePrice(String serial_num, Employee e, double modified_price) throws Exception{
+        // modified_price can be both positive and negative!
+        Item item = TempDataStorage.getItem(serial_num);
+        boolean check_authority = e.getUpdatePrice();
+        if (check_authority){
```

Comment:
I feel checking whether the user has the authority or not would depend on the controller that will call the method and we wouldn't have to do it once the method is called.

Reply:
Also I think if the employee is verified then he/she would only be shown the screen that allows them to choose functions that they're allowed to do, hence we won't have to check it in the method.